packages/lib/pod/src/podUtil.ts

```diff
@@ -353,19 +383,58 @@ export function podEntriesToSimplifiedJSON(
   entries: PODEntries,
   space?: number
 ): string {
-  const simplified: Record<string, bigint | string> = {};
+  const simplified: Record<string, PODRawValue> = {};
   for (const [name, value] of Object.entries(entries)) {
-    simplified[name] = value.value;
+    simplified[name] = podValueToRawValue(value);
   }
   return JSONBig({
     useNativeBigInt: true,
     alwaysParseAsBig: true
   }).stringify(simplified, null, space);
 }
 
+/**
+ * Deserializes `PODValue` from the 'raw value' produced by
+ * {@link podValueToRawValue}.  Type information is inferred from the values
+ * in a way which should preserve hashing and circuit behavior, but isn't
+ * guaranteed to be identical to the types before serialization.  For instance,
+ * small numbers are always annotated as `int`, rather than `cryptographic`.
+ *
+ * @param rawValue a string or bigint representation of `PODValue`
+ * @returns `PODValue` deserialized from the aforementioned value
+ * @throws if the serialized form is invalid
+ */
+export function podValueFromRawValue(rawValue: PODRawValue): PODValue {
+  switch (typeof rawValue) {
+    case "bigint":
+      if (rawValue > POD_INT_MAX) {
+        return { type: "cryptographic", value: rawValue };
+      } else {
+        return { type: "int", value: rawValue };
+      }
+    case "string":
+      return { type: "string", value: rawValue };
+  }
```

Comment:
Without a default case, this is unsafe to use on deserialized data, because you haven't yet checked whether the deserialized object actually has one of the expected types.  That needs to either be handled here (and documented in a @throws clause) or handled by the deserialization code which calls this.